recipes-domu/domu-image-android/domu-image-android.bb

```diff
@@ -1,76 +1,123 @@
 
-###############################################################################
-# extra layers and files to be put after Yocto's do_unpack into inner builder
-###############################################################################
-# these will be populated into the inner build system on do_unpack_xt_extras
-XT_QUIRK_UNPACK_SRC_URI += " \
-    file://meta-xt-images-extra;subdir=repo \
-    file://xt_reconstruct.bbclass;subdir=repo/meta-xt-images-extra/classes \
-"
+# here we handle prebuild related things
+python __anonymous () {
+    # Depending on XT_ANDROID_PREBUILDS_DIR we either use proprietary sources or use prebuilt graphics
+    if d.getVar("XT_ANDROID_PREBUILDS_DIR", expand=True):
+        DDK_KM_PREBUILT_MODULE = "${XT_ANDROID_PREBUILDS_DIR}/pvr-km/pvrsrvkm.ko"
+        DDK_UM_PREBUILDS = "${XT_ANDROID_PREBUILDS_DIR}/pvr-um/"
+    else:
+        # groups=all results in fetching of proprietary projects
+        d.appendVar("ANDROID_SOURCES", ";groups=all")
 
-# these layers will be added to bblayers.conf on do_configure
-XT_QUIRK_BB_ADD_LAYER += " \
-    meta-xt-images-extra \
-"
+    prebuilt_vars = ["DDK_KM_PREBUILT_MODULE", "DDK_UM_PREBUILDS", "TARGET_PREBUILT_KERNEL"]
+    for var in prebuilt_vars:
+        if d.getVar(var, True) != None:
+            if not os.path.exists(d.getVar(var, True)):
+                raise bb.parse.SkipPackage('%s points to a non-existent path' % (var))
+            d.appendVar("PREBUILT_VARS", "%s=%s\n" % (var, d.getVar(var, True)))
+}
 
-SRC_URI_append = " \
-    file://0001-fetch2-repo-Always-check-if-branch-is-correct.patch;patchdir=poky \
-    file://0002-fetch2-repo-Make-fetcher-always-sync-on-unpack.patch;patchdir=poky \
-    file://0003-fetch2-repo-Use-multiple-jobs-to-fetch-and-sync.patch;patchdir=poky \
-    file://0004-Make-pack-unpack-multithreaded.patch;patchdir=poky \
-    file://0005-fetch2-repo-Add-group-parameter.patch;patchdir=poky \
-"
+SRC_URI = "${ANDROID_SOURCES}"
 
-update_local_conf() {
-    local local_conf="${S}/build/conf/local.conf"
+# SOC_FAMILY is required to be set for proper build of optee, graphic and output folder name.
+# See product's append for properly set value
+SOC_FAMILY ?= "SOC_FAMILY_is_not_set"
 
-    cd ${S}
+# SOC_REVISION is used only for build of graphic drivers.
+# See product's append for possible (but not mandatory) override
+SOC_REVISION = ""
 
-    # override what xt-distro wants as machine as we will only use
-    # machines understood by Poky
-    base_update_conf_value ${local_conf} MACHINE "qemux86-64"
 
-    if [ ! -z ${XT_ANDROID_PREBUILDS_DIR} ];then
-        base_set_conf_value ${local_conf} DDK_KM_PREBUILT_MODULE "${XT_ANDROID_PREBUILDS_DIR}/pvr-km/pvrsrvkm.ko"
-        base_set_conf_value ${local_conf} DDK_UM_PREBUILDS "${XT_ANDROID_PREBUILDS_DIR}/pvr-um/"
-    fi
-}
+# Product and variant are used to form string for 'lunch xenvm-userdebug'
+# These two are not suppoesed to be changed.
+ANDROID_PRODUCT = "xenvm"
```

Comment:
Maybe `?=` would be better there and on the next line?

Just in case I'll want to set `ANDROID_VARIANT=user` in my `local.conf`

Reply:
Thanks for pointing.
I have changed assignment to weak default (??=) and provided comment with possible places to redefine these variables.